frontend/public/components/replicaset.jsx

```diff
@@ -25,11 +25,68 @@ import {
 } from './utils';
 import { ResourceEventStream } from './events';
 import { VolumesTable } from './volumes-table';
-import { ReplicaSetModel } from '../models';
+import {
+  DeploymentConfigModel,
+  DeploymentModel,
+  ReplicaSetModel,
+  ReplicationControllerModel,
+} from '../models';
+import { getDeploymentConfigNameForRC, getDeploymentNameForRS } from '@console/shared/src';
+import { rollbackModal } from './modals';
 
 const { ModifyCount, AddStorage, common } = Kebab.factory;
 
+const INACTIVE_STATUSES = ['New', 'Pending', 'Running'];
+
+const RollbackAction = (kind, obj) => {
+  if (kind.kind === ReplicationControllerModel.kind) {
+    const deploymentPhase = obj?.metadata?.annotations?.['openshift.io/deployment.phase'];
+    const dcName = getDeploymentConfigNameForRC(obj);
+    return {
+      // t('public~Rollback')
+      labelKey: 'public~Rollback',
+      // disabled if the DC is not Active
+      isDisabled: INACTIVE_STATUSES.includes(deploymentPhase),
+      // Hidden if RC is active or does not belong to a deployment config
+      hidden: !deploymentPhase || obj?.status?.replicas > 0 || !dcName,
+      callback: () =>
+        rollbackModal({
+          resourceKind: kind,
+          resource: obj,
+        }),
+      accessReview: {
+        group: DeploymentConfigModel.apiGroup,
+        resource: DeploymentConfigModel.plural,
+        name: dcName,
+        namespace: obj?.metadata?.namespace,
+        verb: 'update',
+      },
+    };
+  }
+
+  const deploymentName = getDeploymentNameForRS(obj);
+  return {
+    // t('public~Rollback')
+    labelKey: 'public~Rollback',
+    // Hidden if RS is active or does not belong to a deployment
+    hidden: obj?.status?.replicas > 0 || !deploymentName,
+    callback: () =>
+      rollbackModal({
+        resourceKind: kind,
+        resource: obj,
+      }),
+    accessReview: {
+      group: DeploymentModel.apiGroup,
+      resource: DeploymentModel.plural,
+      name: deploymentName,
+      namespace: obj?.metadata?.namespace,
+      verb: 'update',
```

Comment:
Deployment rollback should check the `patch` verb since we use `k8sPatch`

```suggestion
      verb: 'patch',
```